code/game/objects/items/devices/flashlight.dm

```diff
@@ -121,23 +124,28 @@
 					else
 						M.visible_message(span_notice("[M] directs [src] to [M.p_their()] eyes."), span_notice("You wave the light in front of your eyes."))
 				else
-					user.visible_message(span_warning("[user] directs [src] to [M]'s eyes."), \
-						span_danger("You direct [src] to [M]'s eyes."))
-					if(M.stat == DEAD || (M.is_blind()) || !M.flash_act(visual = 1)) //mob is dead or fully blind
-						to_chat(user, span_warning("[M]'s pupils don't react to the light!"))
-					else if(M.dna && M.dna.check_mutation(/datum/mutation/human/xray)) //mob has X-ray vision
-						to_chat(user, span_danger("[M]'s pupils give an eerie glow!"))
-					else //they're okay!
-						to_chat(user, span_notice("[M]'s pupils narrow."))
+					user.visible_message(span_warning("[user] directs [src] to [M]'s eyes."), ignored_mobs = user)
+					render_list += "<span class='notice'>You direct [src] to [M]'s eyes:</span>\n"
+
+					if(M.stat == DEAD || !M.flash_act(visual = 1))
```

Comment:
Your pr body says `Removes ability to detect blindness with this examination.`, but you still have the ability to detect blindness right there. Flash act returns false if it fails (they are blind).

Reply:
Weird facet about the stat check as well - you are blind when unconscious or in hard crit. So it'll fail on mobs which are unconscious, hard crit, in fake death (regen coma / ling stasis), or dead.

Reply:
(I didn't actually know what flash_act was, and didn't want to break anything ;-;)
Nixed that check and cleaned up the code, should function as advertised now! (and no longer produce a screen flash effect)